pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -210,12 +209,7 @@ func (configgen *ConfigGeneratorImpl) buildOutboundClusters(proxy *model.Proxy,
 			}
 
 			setUpstreamProtocol(proxy, defaultCluster, port, model.TrafficDirectionOutbound)
-			serviceMTLSMode := model.MTLSUnknown
-			if !service.MeshExternal {
-				// Only need the authentication MTLS mode when service is not external.
-				policy, _ := push.AuthenticationPolicyForWorkload(service, port)
-				serviceMTLSMode = authn_v1alpha1_applier.GetMutualTLSMode(policy)
-			}
+			serviceMTLSMode := push.BestEffortInferServiceMTLSMode(service, port)
```

Comment:
I don't really understand this. Why are we replacing something that is correct with something that is "best effort"? Am I missing something here?

Reply:
With the workload-oriented policy, we cannot tell what is the mTLS mode of the`service` anymore. 
It's still correct if beta policy is not used and fallback to alpha.